File: samples/server/petstore/springboot-x-implements-skip/src/main/java/org/openapitools/configuration/EnumConverterConfiguration.java
package org.openapitools.configuration;

import org.openapitools.model.EnumClassDto;
import org.openapitools.model.OuterEnumDefaultValueDto;
import org.openapitools.model.OuterEnumDto;
import org.openapitools.model.OuterEnumIntegerDefaultValueDto;
import org.openapitools.model.OuterEnumIntegerDto;
import org.openapitools.model.SingleRefTypeDto;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.convert.converter.Converter;

/**
 * This class provides Spring Converter beans for the enum models in the OpenAPI specification.
 *
 * By default, Spring only converts primitive types to enums using Enum::valueOf, which can prevent
 * correct conversion if the OpenAPI specification is using an `enumPropertyNaming` other than
 * `original` or the specification has an integer enum.
 */
@Configuration(value = "org.openapitools.configuration.enumConverterConfiguration")
public class EnumConverterConfiguration {

    @Bean(name = "org.openapitools.configuration.EnumConverterConfiguration.enumClassConverter")
    Converter<String, EnumClassDto> enumClassConverter() {
        return new Converter<String, EnumClassDto>() {
            @Override
            public EnumClassDto convert(String source) {
                return EnumClassDto.fromValue(source);
            }
        };
    }
    @Bean(name = "org.openapitools.configuration.EnumConverterConfiguration.outerEnumDefaultValueConverter")
    Converter<String, OuterEnumDefaultValueDto> outerEnumDefaultValueConverter() {
        return new Converter<String, OuterEnumDefaultValueDto>() {
            @Override
            public OuterEnumDefaultValueDto convert(String source) {
                return OuterEnumDefaultValueDto.fromValue(source);
            }
        };
    }
    @Bean(name = "org.openapitools.configuration.EnumConverterConfiguration.outerEnumConverter")
    Converter<String, OuterEnumDto> outerEnumConverter() {
        return new Converter<String, OuterEnumDto>() {
            @Override
            public OuterEnumDto convert(String source) {
                return OuterEnumDto.fromValue(source);
            }
        };
    }
    @Bean(name = "org.openapitools.configuration.EnumConverterConfiguration.outerEnumIntegerDefaultValueConverter")
    Converter<Integer, OuterEnumIntegerDefaultValueDto> outerEnumIntegerDefaultValueConverter() {
        return new Converter<Integer, OuterEnumIntegerDefaultValueDto>() {
            @Override
            public OuterEnumIntegerDefaultValueDto convert(Integer source) {
                return OuterEnumIntegerDefaultValueDto.fromValue(source);
            }
        };
    }
    @Bean(name = "org.openapitools.configuration.EnumConverterConfiguration.outerEnumIntegerConverter")
    Converter<Integer, OuterEnumIntegerDto> outerEnumIntegerConverter() {
        return new Converter<Integer, OuterEnumIntegerDto>() {
            @Override
            public OuterEnumIntegerDto convert(Integer source) {
                return OuterEnumIntegerDto.fromValue(source);
            }
        };
    }
    @Bean(name = "org.openapitools.configuration.EnumConverterConfiguration.singleRefTypeConverter")
    Converter<String, SingleRefTypeDto> singleRefTypeConverter() {
        return new Converter<String, SingleRefTypeDto>() {
            @Override
            public SingleRefTypeDto convert(String source) {
                return SingleRefTypeDto.fromValue(source);
            }
        };
    }

}

File: samples/server/petstore/kotlin-springboot-x-kotlin-implements/src/main/kotlin/org/openapitools/configuration/EnumConverterConfiguration.kt
package org.openapitools.configuration

import org.openapitools.model.Color
import org.openapitools.model.SomeEnum
import org.openapitools.model.SomeNullableEnumWithNonNullDefault
import org.openapitools.model.SomeNullableEnumWithNullDefault

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.core.convert.converter.Converter

/**
 * This class provides Spring Converter beans for the enum models in the OpenAPI specification.
 *
 * By default, Spring only converts primitive types to enums using Enum::valueOf, which can prevent
 * correct conversion if the OpenAPI specification is using an `enumPropertyNaming` other than
 * `original` or the specification has an integer enum.
 */
@Configuration(value = "org.openapitools.configuration.enumConverterConfiguration")
class EnumConverterConfiguration {

    @Bean(name = ["org.openapitools.configuration.EnumConverterConfiguration.colorConverter"])
    fun colorConverter(): Converter<kotlin.String, Color> {
        return object: Converter<kotlin.String, Color> {
            override fun convert(source: kotlin.String): Color = Color.forValue(source)
        }
    }
    @Bean(name = ["org.openapitools.configuration.EnumConverterConfiguration.someEnumConverter"])
    fun someEnumConverter(): Converter<kotlin.String, SomeEnum> {
        return object: Converter<kotlin.String, SomeEnum> {
            override fun convert(source: kotlin.String): SomeEnum = SomeEnum.forValue(source)
        }
    }
    @Bean(name = ["org.openapitools.configuration.EnumConverterConfiguration.someNullableEnumWithNonNullDefaultConverter"])
    fun someNullableEnumWithNonNullDefaultConverter(): Converter<kotlin.String, SomeNullableEnumWithNonNullDefault> {
        return object: Converter<kotlin.String, SomeNullableEnumWithNonNullDefault> {
            override fun convert(source: kotlin.String): SomeNullableEnumWithNonNullDefault = SomeNullableEnumWithNonNullDefault.forValue(source)
        }
    }
    @Bean(name = ["org.openapitools.configuration.EnumConverterConfiguration.someNullableEnumWithNullDefaultConverter"])
    fun someNullableEnumWithNullDefaultConverter(): Converter<kotlin.String, SomeNullableEnumWithNullDefault> {
        return object: Converter<kotlin.String, SomeNullableEnumWithNullDefault> {
            override fun convert(source: kotlin.String): SomeNullableEnumWithNullDefault = SomeNullableEnumWithNullDefault.forValue(source)
        }
    }

}
